README.md

```diff
@@ -1,14 +1,117 @@
-# A2Go web-workshop
-A series of progressive workshops to build webservices
+# Workshop 0
 
-### Our story so far
+In this workshop we will build a simple, basic HTTP server. This assumes you have no experience with Go's net/http library.
 
-We have an app that says hello world!
+## Getting Started
 
-### Client
+Create a file to hold your main function. It doesn't matter where, the root of this repository is fine. Write a main function and try running it with `go run`. Try building it with `go build`. You may wish to output some text to the console so that you know for sure that things are working.
 
-Go to <a href="./client">The client directory</a> for instructions on how to run the client!
+<details><summary>Example</summary>
 
-### Get started with Go
+```go
+package main
 
-The quickest way to get started with Go, is install [VS Code](https://github.com/microsoft/vscode) (Jet Brains GoLand is nice, but not free). If you go to the [vscode-config](./vscode-config) directory, there's useful tips and settings for you to get started with.
+import "fmt"
+
+func main() {
+       fmt.Println("hello world")
+}
+
+```
+
+</details>
+
+## Make an HTTP listener
+
+To make an HTTP server, the `net/http` package needs to be imported. The `ListenAndServe` function in that package starts a TCP listener that responds to HTTP requests.
+
+If you are using Visual Studio Code, you can type `http.ListenAndServe` and press cmd-. to automatically add `net/http` to the imports.
+
+Pick a port on which to listen and pass nil as the handler for now.
+
+<details><summary>Example</summary>
+
+```go
+package main
+
+import (
+    "net/http"
+)
+
+func main() {
+    http.ListenAndServe(":8000", nil)
+}
+```
+
+</details>
+
+Run it again and try to connect with a web browser or curl.
+
+```sh
+curl http://localhost:8000/
+```
+
+Go is generating a 404 response for us. Use `curl -D -` to see the HTTP headers and notice which headers it created for us.
+
+## Add a response
+
+It is always a good idea to read the documentation for functions that we use. One of Go's strengths is the excellently documented standard library. Read https://golang.org/pkg/net/http/#ListenAndServe
+
+We can use something called the default serve mux. We don't go into what this means in this workshop. We save that for a later workshop. Expand the example on the godocs if you would like to see the next step of our code, or use `http.HandleFunc` and write the code yourself.
+
+## Logging the error
+
+One important part of writting Go is to leave no error unchecked, usually. If a function returns an error, that error should probably be part of control flow and maybe logged.
+
+You may have noticed that `http.ListenAndServe` returns an error. It is probably save to ignore any error that comes from it, but what if we had fat fingered our listen port as `";8000"` instead of `":8000"`? What do you think would happen?
```

Comment:
s/save/safe

Reply:
I wouldn't expect an error to come out of ListenAndServe if you type `;8000` . I'd expect the browser wouldn't even try to connect to your port.

Reply:
```suggestion
You may have noticed that `http.ListenAndServe` returns an error. It is probably safe to ignore any error that comes from it, but what if we had fat fingered our listen port as `";8000"` in our code instead of `":8000"`? What do you think would happen?
```